topics/dev/tutorials/debugging/tutorial.md

```diff
@@ -0,0 +1,730 @@
+---
+layout: tutorial_hands_on
+
+title: "Debugging Galaxy"
+questions:
+  - "How do I debug Galaxy?"
+objectives:
+  - "Fix a broken branch"
+  - "Interpret the results of failed tests on GitHub"
+  - "Run individual tests locally"
+  - "Fix errors identified by failing tests"
+  - "Debug simple runtime errors using the Python debugger"
+  - "Write tests exposing the identified bug"
+time_estimation: "4h"
+key_points:
+  - "TODO Unit Tests"
+  - "TODO API Tests"
+  - "TODO Client Linting Failures"
+  - "TODO Client Tests"
+  - "TODO Selenium Tests"
+  - "TODO Runtime Errors"
+contributors:
+  - assuntad23
+  - ic4f
+  - jmchilton
+requirements:
+  - type: none
+    title: "Familiarity with basic Git commands"
+  - type: none
+    title: "Basic knowledge of Python and JavaScript"
+  - type: none
+    title: "Mac OS or Linux that can run Galaxy & your favorite IDE or editor"
+  - type: "internal"
+    topic_name: dev
+    tutorials:
+      - architecture
+subtopic: core
+---
+
+# Introduction
+{:.no_toc}
+
+
+In this tutorial we will demonstrate how to find and fix common types of bugs you may encounter as a contributor to Galaxy. We will step you through the process of finding and fixing a bug - from locating specific errors in the logs of Galaxy's GitHub workflows, to identifying their cause, developing a solution and committing your edits.
+
+With the skills from this tutorial, it is our hope that you will feel more prepared to develop solutions for Galaxy and more confidently navigate any obstacles along the way.
+
+> ### Agenda
+>
+> 1. TOC
+> {:toc}
+>
+{: .agenda}
+
+# Local development environment setup
+
+For almost every type of error you encounter, there will be a relatively similar process:
+
+1. Locate failing test or error on GitHub workflows
```

Comment:
maybe github actions?

Reply:
If this is more appropriate/easy to understand, yes, sure! (was my idea to call them workflows - and I was not at all sure that was a good term in this context). Feel free to suggest the necessary edits - there may be more than one instance.